### PR TITLE
Refactor: subscription controller to allow upgrade

### LINKED ISSUE #502: [FEATURE REQUEST]: Update subcription endpoint to allow subscription upgrade.

### PR DESCRIPTION
**Pull request by oscar-ekeyekwu**
**Before submitting your PR for review**

- Run `npm run lint` to find errors in code syntax/format
- Run `npm run lint:fix` to fix all auto-fixable errors in source code and auto-format with prettier
- Ensure you fix any linting errors displayed after running any of the above commands

**What does this PR do?**

Fixes #502   (replace issue_number with the issue number this PR solves)

- [x] Modify subscription controller to allow subscription upgrade

**description of Task to be completed?**

**How should this be manually tested?**

Give bullet point instructions on how to setup the code and test it manually (Assume no prior experience on JS or code)

**Any background context you want to provide?**

Any pertinent information that should be considered

**What is the link to the issue on Github?**

[Issue #502 ](https://github.com/microapidev/comment-microapi/issues/502)

**Questions:**

If something is unclear or you want some questions to be addressed by your peers, mention them here

### CODE REVIEW

File: controllers/subscriptionsController/subscribeSingleApplication.js [hunk 1]
```diff
@@ -47,89 +83,143 @@ const subscribeSingleApplication = async (req, res, next) => {
...
+        201,
+        isSubscribed,
+        "Application subscription upgrade successful"
+      );
+    } else {
+      //create subscription data & properties objects
+      //logging object
+      const logging = {
+        value: plan.logging,
+        maxLogRetentionDays: plan.maxLogRetentionPeriod,
```
Comment (dave-ok) [inline controllers/subscriptionsController/subscribeSingleApplication.js:155]: I think we should stick with one here. maxLogRetentionsPerDay maybe for consistency?

File: controllers/subscriptionsController/subscribeSingleApplication.js [hunk 2]
```diff
@@ -47,89 +83,143 @@ const subscribeSingleApplication = async (req, res, next) => {
       next(new CustomError(404, "Plan not found!"));
       return;
     }
+
     //calculate subscription expiry date
     const totalPeriod = parseInt(parseInt(plan.periodWeight) * periodCount, 10);
```
[the remark this answers is not in the document]
Comment (oscar-ekeyekwu) [reply controllers/subscriptionsController/subscribeSingleApplication.js:88]: fixed

File: controllers/subscriptionsController/subscribeSingleApplication.js [hunk 3]
```diff
@@ -47,89 +83,143 @@ const subscribeSingleApplication = async (req, res, next) => {
...
-    //populate subscription History Data
-    const subscriptionHistoryData = {
+    isSubscribed = await SubscriptionsModel.findOne({
       applicationId: applicationId,
-      planId: planId,
-      period: `${totalPeriod} months`,
-      expiresOn: expiryDate,
-      subscribedOn: subscriptionDate,
-    };
+    }).sort({ createdAt: "desc" });
```
[the remark this answers is not in the document]
Comment (oscar-ekeyekwu) [reply controllers/subscriptionsController/subscribeSingleApplication.js:96]: fixed

File: controllers/subscriptionsController/subscribeSingleApplication.js (hunk 1 above)
Comment (oscar-ekeyekwu) [reply controllers/subscriptionsController/subscribeSingleApplication.js:155]: fixed

**Pull request merged**